#161255356 - Social login

Jwesh opened the pull request:
#### What does this PR do?
* Enable Social login via Google, Facebook and Twitter
#### Description of Task to be completed?
* Currently a new user cannot login with their pre-existing social media accounts. He/She can only register and or login with email password combinations
#### How should this be manually tested?
1. clone the repo and switch to ft-social-login-161255356
2. set environment and also configure database 
3. get token from google [developer playground](https://developers.google.com/oauthplayground/)
4. get token from facebook [developer console](https://developers.facebook.com/apps/541475206299733/roles/test-users/)

#### Any background context you want to provide?
* social login improves user experience while using the application.
#### What are the relevant pivotal tracker stories?
* Enable Social login via Google, Facebook and Twitter-[161255356](https://www.pivotaltracker.com/story/show/161255356)
#### Screenshots
<img width="964" alt="screen shot 2018-11-01 at 14 13 19" src="https://user-images.githubusercontent.com/17101765/47853392-9fa34600-ddef-11e8-9693-fbf072c7c9b6.png">

Commit:
Feature(social login): add social login

- ensure user can login with facebook, google and twitter
- added a method to generate or to encode JWT token
- updated requirements.txt

[Feature #161255356]

Commit:
Feature(social login): add social login

- ensure user can login with facebook, google and twitter
- added a method to generate or to encode JWT token
- updated requirements.txt

[Feature #161255356]

Commit:
Feature(social login): add social login tests

    - ensure user cannot login without a valid provider and token
    - added test_social_auth which has all the test
    - updated urls to include namespace

    [Feature #161255356]

Commit:
Feature(social login): add social login tests

    - ensure user cannot login without a valid provider and token
    - added test_social_auth which has all the test
    - updated urls to include namespace

    [Feature #161255356]

Commit:
Feature(social login): add social login tests

        - update .travis.yml with postgres v 9.4

        [Feature #161255356]

Commit:
Feature(social login): add social login tests

        - update .travis.yml with postgres v 9.4

        [Feature #161255356]

AnguleMathias reviewed (approved):
Looks great! 
Check on test coverage.

meshnesh reviewed (changes requested):
Write test for the added code to increase the coverage

Commit:
Feature(social login): add social login tests

        - added social login test for twitter
       - solve issues with twitter social auth

        [Feature #161255356]

Commit:
Feature(social login): add social login tests

        - added social login test for twitter
       - solve issues with twitter social auth

        [Feature #161255356]

Commit:
Feature(social login): add social login tests

            - fix merge conflict

            [Feature #161255356]

Commit:
Feature(social login): add social login

- integrating changes from members

[Feature #161255356]

maxwellgithinji reviewed (approved):
LGTM

meshnesh reviewed (changes requested):
Fix the failing test.

meshnesh commented on authors/apps/authentication/backends.py:
```diff
@@ -12,58 +12,71 @@
+        email andress
+        """
+        payload = {
+            'email': email,
```
Is the token supposed to be encoded with the `email` or `user_id`?

Commit:
Feature(social login): ensure jwt is encoded using userid

                - encode jwt using user_id

                [Feature #161255356]
